src/hotspot/share/jvmci/jvmciCodeInstaller.cpp

```diff
@@ -813,14 +816,12 @@ JVMCI::CodeInstallResult CodeInstaller::install(JVMCICompiler* compiler,
         DirectivesStack::release(directive);
       }
 
-      if (nm != nullptr) {
-        if (_nmethod_entry_patch_offset != -1) {
-          err_msg msg("");
-          BarrierSetNMethod* bs_nm = BarrierSet::barrier_set()->barrier_set_nmethod();
+      if (_nmethod_entry_patch_offset != -1) {
+        BarrierSetNMethod* bs_nm = BarrierSet::barrier_set()->barrier_set_nmethod();
 
-          if (!bs_nm->verify_barrier(nm, msg)) {
-            JVMCI_THROW_MSG_(IllegalArgumentException, err_msg("nmethod entry barrier is malformed: %s", msg.buffer()), JVMCI::ok);
-          }
+        err_msg msg("");
```

Comment:
delete `err_msg msg("");`

Reply:
It's necessary to allocate an empty error buffer for use by the verify_barrier code.  Is there some better way to express this?

Reply:
Oh, I missed that `msg` is used below. I don't know of a better way to do this. Maybe just add a comment:
```
// an empty error buffer for use by the verify_barrier code
```

Reply:
ok